src/priorityq.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding utf-8 -*-
+
+
+class Priorityq(object):
+
+    def __init__(self, iterable=None):
+        '''Initialise a Priorityq from a list of lists or tuples
+            where the zeroeth element is the priority'''
+        if iterable is None:
+            self.priorityq = None
+        else:
+            try:
+                self.priorityq = sorted(iterable, key=lambda tup: tup[0])
+            except TypeError:
+                raise TypeError('please enter a list/tuple of '
+                                'lists/tuples with priority as '
+                                'the first value')
+
+    def insert(self, item):
+        ''' inserts an item into the queue.'''
+        if item is None:
```

Comment:
First off, your function docstring does not specify that `item` should be be a tuple, so the user would have no way of knowing.

It would be more Pythonic for you to have `insert` require two arguments, `value` and `priority`, and handle any packing of them into a tuple within the function.